README.md

```diff
@@ -1,31 +1,348 @@
 # Hyperion
 
-TODO: Write a gem description
+Hyperion is a Ruby REST client that follows certain conventions
+layered on top of HTTP. The conventions implement best practices
+surrounding versioning, error handling, and crafting an API for
+consumption by third parties. Hyperion provides abstractions that
+make it easy to follow these conventions consistently across
+projects.
 
-## Installation
+This document describes the conventions, then demonstrates how to
+use the API, and finally shows how to test your client-side code.
 
-Add this line to your application's Gemfile:
+## Conventions
+
+### Versioning
+
+_TODO_
+
+### Errors
+
+400-level errors are always returned as exactly 400; no distinction is
+made in the error code. Instead, a well-defined structure is returned
+that contains
+
+- a human-oriented error message, and
+- a machine-oriented list of `ErrorInfo`s.
```

Comment:
Fixed it to be more abstract about "error info" rather than `ErrorInfo`. I shouldn't be mentioning code concepts here. So I'm not adding a link. It would be brittle anyway.